assignment1/src/main.rs

```diff
@@ -1,35 +1,194 @@
-use std::{fs::File, io::Write};
+
+use std::{env, fs::File, io::Write, path::Path};
+use hound::{WavReader, WavWriter, WavSpec, SampleFormat};
 
 mod comb_filter;
+use comb_filter::{CombFilter, FilterType};
 
 fn show_info() {
     eprintln!("MUSI-6106 Assignment Executable");
     eprintln!("(c) 2024 Stephen Garrett & Ian Clester");
 }
 
 fn main() {
-   show_info();
+    show_info();
 
     // Parse command line arguments
-    let args: Vec<String> = std::env::args().collect();
-    if args.len() < 3 {
-        eprintln!("Usage: {} <input wave filename> <output text filename>", args[0]);
-        return
+    let args: Vec<String> = env::args().collect();
+    if args.len() == 0 {
+        println!("No command line arguments provided, running tests...");
+        test_fir_output_zero_on_feedforward_freq();
+        test_iir_magnitude_change_on_feedback_freq();
+        test_varying_input_block_size();
+        test_processing_zero_input_signal();
+        test_buffer_length_greater_than_input_length();
+        std::process::exit(1);
+    }
```

Comment:
I find it very clever how you implemented your test. This seems very useful when you don't have a file to input. I wonder if there is a way to add an if statement that only runs this block of code when the program is a development build, so that way when creating shipping builds, you don't have to remove it. I understand that what cargo test is for, but I like the way you've done it!